app/helpers/application_helper/button/storage_service_new.rb

```diff
@@ -0,0 +1,21 @@
+class ApplicationHelper::Button::StorageServiceNew < ApplicationHelper::Button::Basic
+  def calculate_properties
+    super
+    if disabled?
+      self[:title] = _("No storage providers support creating storage services.")
+    end
+  end
+
+  # If we selected specific EMS disable create button if not supporting create action.
+  # Else disable button if no active providers support create action
+  def disabled?
+    if @view_context.params["id"]
+      current_ems = ExtManagementSystem.where(:id => @view_context.params["id"]).first
```

Comment:
```suggestion
      current_ems = ExtManagementSystem.find(@view_context.params["id"])
```

or If you don't want to raise an exception if we can't find by ID
```suggestion
      current_ems = ExtManagementSystem.find_by(:id => @view_context.params["id"])
```
but based on the next line I think you want to raise if it can't find one since you're not checking for `nil`